src/gui/qgsstatusbarmagnifierwidget.cpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+                         qgsstatusbarmagnifierwidget.cpp
+    begin                : April 2016
+    copyright            : (C) 2016 Paul Blottiere, Oslandia
+    email                : paul dot blottiere at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QFont>
+#include <QHBoxLayout>
+#include <QLabel>
+
+#include <qgsapplication.h>
+#include "qgsstatusbarmagnifierwidget.h"
+#include "qgsmapcanvas.h"
+#include "qgsdoublespinbox.h"
+
+QgsStatusBarMagnifierWidget::QgsStatusBarMagnifierWidget( QWidget* parent,
+    QgsMapCanvas *canvas ) :
+    QWidget( parent ),
+    mCanvas( canvas ),
+    mMagnifier( 1.0 ),
+    mMagnifierMin( 1.0 ),
+    mMagnifierMax( 10.0 )
+{
+  // label
+  mLabel = new QLabel( this );
+  mLabel->setMinimumWidth( 10 );
+  mLabel->setMargin( 3 );
+  mLabel->setAlignment( Qt::AlignCenter );
+  mLabel->setFrameStyle( QFrame::NoFrame );
+  mLabel->setText( tr( "Magnifier" ) );
+  mLabel->setToolTip( tr( "Magnifier" ) );
+
+  mSpinBox = new QgsDoubleSpinBox( this );
+  mSpinBox->setClearValue( mMagnifierMin );
+  mSpinBox->setKeyboardTracking( false );
+  mSpinBox->setMaximumWidth( 120 );
+  mSpinBox->setDecimals( 1 );
+  mSpinBox->setRange( mMagnifierMin, mMagnifierMax );
+  mSpinBox->setWrapping( true );
```

Comment:
@nyalldawson
Considering that it will be a non pixelated magnification, unlike what it was initially expected, I think it's indeed a good idea to increase the maximum larger! Ditto for the % instead of the magnification factor.